feat: add fortune telling app (#62167)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/blocks/workshop-fortune-teller-app/68ee6bb2b7ed9654f2b7ae8b.md

---
id: 68ee6bb2b7ed9654f2b7ae8b
title: Step 28
challengeType: 0
dashedName: step-28
---

# --description--

Add another event listener to the global document object that calls the `showFortune` function. Inside the callback, include a parameter e of type Event, pass `e` as argument to showFortune function.

# --hints--

You should have assigned `click` event listener to the global document.

```js
assert.match(code, /document\s*\.\s*addEventListener\s*\(\s*['"]click['"]/);
```

Your event listener should have callback parameter named `e`.

```js
assert.match(code, /addEventListener\s*\([\s\S]*?\(\s*e\s*[:)]/);
```

> Your callback parameter should have type of `Event`.

Reviewer: ```suggestion
Your callback parameter should have the type `Event`.
```

```js
assert.match( code, /\(\s*e\s*:\s*Event\s*\)/);
```

> Inside the callback you should have called the `showFortune` function.

Reviewer: ```suggestion
Inside the callback, you should call the `showFortune` function
```

```js
assert.match(code, /addEventListener[\s\S]*?\(\s*[^)]*\)\s*=>\s*(\{[\s\S]*?this\.showFortune\s*\(|this\.showFortune\s*\()/);
```

Inside `showFortune` function you should have passed argument of `e`.

```js
assert.match(code, /showFortune\s*\(\s*e\s*\)/);
```

# --seed--

## --seed-contents--

```html

> <!doctype html>

Reviewer: ```suggestion
<!DOCTYPE html>
```
